receiver/receivercreator/README.md

```diff
@@ -439,3 +439,296 @@ service:
 
 The full list of settings exposed for this receiver are documented [here](./config.go)
 with detailed sample configurations [here](./testdata/config.yaml).
+
+
+## Generate receiver configurations from provided Hints
+
+Note: When hints feature is enabled if hints are present for an endpoint no receiver templates will be evaluated.
+
+Currently this feature is only supported for K8s environments and the `k8sobserver`.
+
+The metrics' hints feature for K8s is enabled with the following setting:
+
+```yaml
+receiver_creator/metrics:
+  watch_observers: [ k8s_observer ]
+  hints:
+    k8s:
+      metrics:
+       enabled: true
+```
+
+The logs' hints feature for K8s is enabled with the following setting:
+
+```yaml
+receiver_creator/logs:
+  watch_observers: [ k8s_observer ]
+  hints:
+    k8s:
```

Comment:
Is the `k8s` configuration key needed? Somehow, I feel like we are creating a configuration relationship between the `watch_observers` (interface) and the `hints` configuration.

Would it make sense to document that `hints` is only supported for k8s at the moment? If in the future can we have other "hints/annotations" observers, they could reuse the same "hints" config (or not):

```yaml
// reusing hints config across observers
receiver_creator/logs:
  watch_observers: [ k8s_observer, file_observer ]
  hints:
    metrics:
      enabled: true

// different configs
receiver_creator/logs/k8s:
  watch_observers: [ k8s_observer ]
  hints:
    metrics:
      enabled: true
receiver_creator/logs/file:
  watch_observers: [ file_observer ]
  hints:
    metrics:
      enabled: true
```

Reply:
The idea is that we can configure the hints feature for multiple environments (in the future) like docker, k8s etc in a way that the configuration can be specific per environment.

So we could have sth like the following:

```yaml
// different configs
receiver_creator/metrics:
  watch_observers: [ k8s_observer, docker_observer ]
  hints:
    k8s: 
      metrics:
        enabled: true
    docker:
      metrics:
        # docker specific settings
```

Happy to reconsider this though if it doesn't make sense.

> Would it make sense to document that hints is only supported for k8s at the moment?

It is mentioned at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35617/files#diff-4127365c4062a7510fb7fede0fa239e9232549732898303d94c12fef0433d39dR448

Reply:
I agree that the nested configuration seems confusing. Users can have separate `receiver_creator` instances with different `watch_observers` if they want to split the behavior between docker/k8s/etc.

I'm also not sure about using the _hints_ term everywhere. It seems not very descriptive to me. What about using `discovery` term instead, given that this is something we already use in the annotations?

Maybe something like this:
```yaml
receiver_creator:
  watch_observers: [k8s_observer]
  discovery:
    enabled: true
    ignore_receivers: [filelog]
```

This would mean only allow creation of metrics related receivers from the pod annotations. Also, if a user finds any issues with let's say `nginx` metrics, they can easily add it to `ignore_receivers`...

Reply:
Cool, it seems we can unify then. I also like the idea of `ignore_receivers` since it provides the option to Collector admins  to disable specific receivers for security reasons. I will update the PR accordingly.